api/v1alpha1/apidefinition_types.go

```diff
@@ -163,8 +170,34 @@ type URLRewriteMeta struct {
 	// MatchPattern is a regular expression pattern to match the path
 	MatchPattern string `json:"match_pattern"`
 	// RewriteTo is the target path on the upstream, or target URL we wish to rewrite to
-	RewriteTo string           `json:"rewrite_to"`
-	Triggers  []RoutingTrigger `json:"triggers"`
+	// +optionall
+	RewriteTo string `json:"rewrite_to,omitempty"`
+	// RewriteToInternal serves as rewrite_to but used when rewriting to target
+	// internal api's
+	// When rewrite_to and rewrite_to_loop are both provided then
+	// rewrite_to will take rewrite_to_loop
+	// +optionall
+	RewriteToInternal *LoopInternal `json:"rewrite_to_internal,omitempty"`
+	// +optionall
+	Triggers []RoutingTrigger `json:"triggers,omitempty"`
+}
+
+// LoopInternal defines options that constructs a url that refers to an api that
+// is loaded into the gateway.
+type LoopInternal struct {
+	// Self set this to true if the api definition is looping to itself.
+	Self bool `json:"self,omitempty"`
```

Comment:
For MVP - Let's not include `Self`, until we have concrete use-case / example and can create a BDD feature which clearly describes and tests that use-case?

Reply:
done bc8072c